bugsnag-android-core/src/androidTest/java/com/bugsnag/android/AppDataTest.java

```diff
@@ -63,20 +63,4 @@ public void testAccessors() {
         assertEquals(500L, appData.get("durationInForeground"));
         assertTrue((Boolean) appData.get("inForeground"));
     }
-
-    @Test
-    public void testJsonSerialisation() throws JSONException {
-        appData.put("buildUUID", "fa54de");
-        JSONObject appDataJson = mapToJson(appData);
-
-        assertEquals(1, appDataJson.getInt("versionCode"));
-        assertEquals("1.0", appDataJson.get("version"));
-        assertEquals("development", appDataJson.get("releaseStage"));
-        assertEquals("android", appDataJson.get("type"));
-        assertEquals("com.bugsnag.android.core.test", appDataJson.get("id"));
-        assertNotNull(appDataJson.get("buildUUID"));
```

Comment:
Do we need to check the serialization of `buildUUIID`, `duration`, `durationInForeground` and `inForeground` in the new tests like this test was?

Reply:
It feels like we have a lot of different test classes - AppDataSummaryTest, AppDataTest, AppDataOverrideTest and AppDataSerialisationTest - but they don't seem to cover all the fields between them.

Can these be rationalised and make sure that we include each of the fields of App?

Reply:
I have removed the various different test classes in favour of `AppDataSerializationTest`, which now covers the serialization of each value defined within AppData.

Reply:
This is also an issue with tests for `DeviceData` - I'll address that in a separate PR.

Reply:
Should we not also have basic unit tests for the accessors (separate from serialisation)? They would be a unit test, not an instrumentation test, but would check the getters return an appropriate value.

Reply:
`AppData#getAppDataSummary()` returns a `Map`, so I'm not sure that testing the accessors would be necessary as we're not defining any of our own. Unless I've got the wrong end of the stick here?